barber/barber.go

```diff
@@ -93,7 +93,7 @@ func (b *barber) IsAvailable(serverID int, tm time.Time) bool {
 		return false
 	}
 
-	res := h.countFails(tm.Unix(), b.config.Threshold)
+	res := h.countFails(tm.Unix(), b.config.MaxFails)
```

Comment:
I think you should write tests for this behavior. Looks like I did a mistake some time ago :D